### DISCUSSION TITLE
Unable to send out invites

### DISCUSSION THREAD

**Discussion by martinl88** [outline/outline#2343] [Self Hosting]
While trying to send out invites, UI says "invites sent" but in reality nothing seems to happen. In docker logs I can see this:

```
outline     | (node:59) UnhandledPromiseRejectionWarning: Error: 140097387232584:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
outline     |
outline     | (node:59) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

I've tried with two different SMTP servers, with and without auth.

**To Reproduce**
1. Click on 'Invite people...'
2. Fill out the form
3. Click on 'Send invites'

**Expected behavior**
Invites should be sent out

**Outline (please complete the following information):**
- Install: self hosted
- Version: v0.57.0
- Docker image: outlinewiki/outline:latest

**Comment by tommoor** [comment]
Try setting the environment variable `SMTP_TLS_CIPHERS=SSLv3`

**Comment by soulteary** [comment]
> Try setting the environment variable `SMTP_TLS_CIPHERS=SSLv3`

Perhaps supporting the use of the API Email service can solve many such problems.

For example, eg. mailgun, or a service unique to each user area.

**Comment by tommoor** [comment]
If there is a bug to fix here it would be that it would be nice to capture this specific error and display a friendly error message to modify the configuration.

**Comment by martinl88** [comment]
> Try setting the environment variable `SMTP_TLS_CIPHERS=SSLv3`

Sadly it didn't help. The same error is being spat out.

**Comment by martinl88** [comment]
OK. I'm getting somewhere. Turns out neither STARTTLS or sending without encryption isn't supported and result in the error stated above. When I used regular SSL/TLS, then the invites started moving.

So, the new question is, how can I configure the encryption part of the SMTP connection?

**Reply by Limezy** [reply]
Did you try `SMTP_TLS_CIPHERS=TLSv1.2` ?
The below configuration worked for me :

```
SMTP_HOST=maildomain.tld
SMTP_PORT=587
SMTP_USERNAME=username
SMTP_PASSWORD=password
SMTP_FROM_EMAIL=outline@maildomain.tld
SMTP_REPLY_EMAIL=webmaster@maildomain.tld
SMTP_TLS_CIPHERS=TLSv1.2
SMTP_SECURE=false
```

**Reply by its-rku** [reply]
Yes, it is working for me too. Thanks  buddy.

**Reply by mm3906078** [reply]
> Did you try `SMTP_TLS_CIPHERS=TLSv1.2` ? The below configuration worked for me :
> 
> ```
> SMTP_HOST=maildomain.tld
> SMTP_PORT=587
> SMTP_USERNAME=username
> SMTP_PASSWORD=password
> SMTP_FROM_EMAIL=outline@maildomain.tld
> SMTP_REPLY_EMAIL=webmaster@maildomain.tld
> SMTP_TLS_CIPHERS=TLSv1.2
> SMTP_SECURE=false
> ```

Hi,
are you sure this is safe?

**Reply by Limezy** [reply]
Well... It's "TLSv1.2 safe" I guess. Not as good as v1.3 but still used by a lot of MTA and clients I think.
Anyway, when it comes to emails, you have to consider that nothing is safe unless PGP encrypted.

**Reply by binaryfire** [reply]
@tommoor I think it might be worth adding "STARTTLS isn't supported" to the docs. I was trying to use Amazon SES over port 587 (STARTTLS) and was getting the same error mentioned here. As soon as I switched to port 465 (regular TLS) it worked.

Ref: https://docs.aws.amazon.com/ses/latest/dg/smtp-connect.html

**Reply by begemotik** [reply]
> SMTP_TLS_CIPHERS=TLSv1.2
SMTP_SECURE=false

I think you actually never use TLS if you set `SMTP_SECURE=false`

Thanks @binaryfire for your finding, it really helped me to setup notification with SES.

**Reply by sourcecodemage** [reply]
> @tommoor I think it might be worth adding "STARTTLS isn't supported" to the docs. I was trying to use Amazon SES over port 587 (STARTTLS) and was getting the same error mentioned here. As soon as I switched to port 465 (regular TLS) it worked.
> 
> Ref: https://docs.aws.amazon.com/ses/latest/dg/smtp-connect.html

Nice. Did you have any other settings? Like "SMTP_TLS_CIPHERS"   or "SMTP_SECURE" ?

**Reply by binaryfire** [reply]
@majorgear Nope, just this:

 ```
SMTP_HOST=email-smtp.eu-central-1.amazonaws.com
SMTP_PORT=465
SMTP_USERNAME={{ aws_ses_username }}
SMTP_PASSWORD={{ aws_ses_password }}
SMTP_FROM_EMAIL=docs@mydomain.dev
SMTP_REPLY_EMAIL=noreply@mydomain.dev
 ```

**Reply by sourcecodemage** [reply]
Thx. My use case was for Apple / iCloud mail. I discovered that this was covered in their wiki. 
[](https://wiki.generaloutline.com/s/hosting/doc/smtp-cqCJyZGMIB#h-apple-mail)

It's a bit different than yours, but I appreciate your response. 

```
SMTP_HOST=smtp.mail.me.com
SMTP_PORT=587
SMTP_USERNAME=<username>
SMTP_PASSWORD=<password>
SMTP_FROM_EMAIL=<email>
SMTP_TLS_CIPHERS=TLSv1.2
SMTP_SECURE=false
```

**Reply by Sleepful** [reply]
They still haven't added "STARTTLS isn't supported" to the docs 🤦 
Took me a long handful of hours to find out the hard way...

**Comment by thetractor** [comment]
I try to get this working with Postmark. Non of the above solutions helped getting things working. Here my configuration:

```
SMTP_HOST=smtp.postmarkapp.com
SMTP_PORT=587
SMTP_USERNAME=<API-TOKEN>
SMTP_PASSWORD=<API-TOKEN>
SMTP_FROM_EMAIL=email@example.com
SMTP_REPLY_EMAIL=email@example.com
```

I tried with ports 25, 587 and 465 (which is not supported according to postmark's docs). Also triede many different settings for the `SMTP_TLS_CIPHERS ` variable - no luck. 

The error I am getting is this:

```
app-outline-1       | {"error":"38DB86990D7F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:\n","level":"error","message":"Error sending email to email@example.com","stack":"Error: 38DB86990D7F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:354:\n"}
```

Any suggestions?

**Reply by thetractor** [reply]
I have now a working configuration, just in case some else wants to setup outline with postmark. My problem was that `docker compose restart` won't refresh the ENV vars inside the container 🤦‍♂️ 

```
SMTP_HOST=smtp.postmarkapp.com
SMTP_PORT=587
SMTP_USERNAME=<API-TOKEN>
SMTP_PASSWORD=<API-TOKEN>
SMTP_FROM_EMAIL=email@example.com
SMTP_REPLY_EMAIL=email@example.com
SMTP_SECURE=false
```

Maybe this can be added to the docs at https://docs.getoutline.com/s/hosting/doc/smtp-cqCJyZGMIB

**Reply by sim2kid** [reply]
It would be nice to know a solution that does use encryption of some sort for Postmark. Turning it off is more like a workaround than a solution.

**Reply by Sleepful** [reply]
Postmark doesn't offer TLS, only STARTTLS, so you are out of luck with Outline if your provider is Postmark, which is a shame. If you have your own Postfix server with a relay to Postmark, you could do that... but it might be easier to just add STARTTLS to Outline lol

**Reply by Sleepful** [reply]
And not using encryption is a really iffy thing to do 🤦

**Reply by tommoor** [reply]
`STARTTLS` has always been supported by Outline, we' just use `nodemailer` as the module for sending emails – it's a very straight forward implementation

**Reply by Sleepful** [reply]
@tommoor is that written somewhere in the docs? Would be good to add. I tried to use it and failed to do so 😄

**Reply by tommoor** [reply]
Happy to add something [to the docs](https://docs.getoutline.com/s/hosting/doc/smtp-cqCJyZGMIB) if you have a suggestion for what would help here.

**Reply by namnguyenthanhwork** [reply]
hi @tommoor, how to setting nodemailer using test sending emails on dev mode, i just add

```
SMTP_HOST=smtp.ethereal.email
SMTP_PORT=587
SMTP_USERNAME=test
SMTP_PASSWORD=test
SMTP_FROM_EMAIL=hello@example.com
SMTP_REPLY_EMAIL=hello@example.com
```

in `.env`, but it not working. Can you help me? Thanks.

[Updated]: I setup sending emails successfully. However, when i create or update docs, no email sended. How to test it? Thanks

**Reply by tommoor** [reply]
> when i create or update docs, no email sended. How to test it?

You would need to have another user subscribed to the document, you won't get emails for your own activity
